File: engine/effects/handlers/movement.py
"""Movement effect handler scaffold."""

from __future__ import annotations

from engine.effects.base import Effect


def handle_movement(effect: Effect) -> dict:
    return {
        "status": "ok",
        "kind": "movement",
        "target": effect.target_unit_id,
        "payload": effect.payload,
    }

File: src/effects/handlers/movement.ts
/**
 * Movement effect handler scaffold.
 * Mirrors engine/effects/handlers/movement.py
 */

import { Effect } from "../base";

export function handleMovement(effect: Effect): Record<string, unknown> {
  return {
    status: "ok",
    kind: "movement",
    target: effect.targetUnitId,
    payload: effect.payload,
  };
}
